samza-test/src/main/config/join/README

```diff
@@ -1,77 +1,86 @@
   When it has received an emitted value from each partition it outputs the key to the topic "completed".
-  To track which partitions have emitted their value it keeps a store with | seperated numbers. 
+  To track which partitions have emitted their value it keeps a store with | seperated numbers.
   The first entry is the epoch and the remaining entries are partitions that have emitted the key.
-  
+
 checker.samza:
-  This job has a single partition and stores all the completed keys. When all the keys are completed it sends an incremented epoch to the epoch topic, kicking off a new round.
-  
+  This job has a single partition and stores all the completed keys.
+  When all the keys are completed it sends an incremented epoch to the epoch topic, kicking off a new round.
+
 watcher.samza:
   This job watches the epoch topic. If the epoch doesn't advance within some SLA this job sends an alert email.
-  
-The state maintained by some of these jobs is slightly complex because of the need to make everything idempotent. So, for example, instead of keeping the partition count
-in the joiner job we keep the set of partitions so that double counting can not occur.
 
-To run, simply start all four jobs at once.
+The state maintained by some of these jobs is slightly complex because of the need to make everything idempotent.
+So, for example, instead of keeping the partition count in the joiner job we keep the set of partitions
+so that double counting can not occur.
 
-* How to setup test ? *
+* How to setup this test? *
 
-NOTE: You will need to have Paramiko installed in order to run these tests.
+You need to have Paramiko installed in order to run these tests. (pip install paramiko)
+This test is meant to be used with hello-samza's bin/grid script (https://github.com/apache/samza-hello-samza).
 
-This test is meant to be used with hello-samza's bin/grid script. If you use hello-samza, and bin/grid bootstrap, then DEPLOY_DIR can be set to samza-hello-samza/deploy/samza,
-and you can run the integration tests with ./bin/setup-int-test.sh path/to/samza-hello-samza/deploy.
+First, set up a few environment variables:
+> export SAMZA_SRC=/path/to/samza
+> export HELLO_SAMZA_SRC=/path/to/samza-hello-samza
+> export DEPLOY_DIR=$HELLO_SAMZA_SRC/deploy
 
-The steps to setup the integration tests is similar to the samza-hello-samza setup, using the grid script.
-Once you deploy Zookeeper, YARN and Kafka, you have to generate the tar ball with the tests.
-Before generating the tar ball, update the "yarn.package.path" in $SAMZA_SRC/src/config/join/common.properties file to the path where the published tar will be made available.
+Deploy Zookeeper, YARN and Kafka:
+> cd $HELLO_SAMZA_SRC
+> bin/grid bootstrap
 
---> Release the tar
-cd $SAMZA_SRC
-./gradlew releaseTestJobs
-cp ~/samza-test/build/distributions/samza-test_*.tgz $DEPLOY_DIR
-tar -xvf $DEPLOY_DIR/samza-test_*.tgz -C $DEPLOY_DIR/samza
+Generate the tarball with the test jobs:
```

Comment:
Does Line 47 serve as title for lines 48 - 51 or 48 - 58? You are not actually generating any tar until Line 55. Kind of confusing.